### PR TITLE
feat: add algolia search

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----
Work in progress to allow for better search based on the Algolia index.
CC @Bo98 who might be working on something similar

### CODE REVIEW

**Comment by Bo98** [comment]
See previous PR here: https://github.com/Homebrew/brew/pull/13520.

However since then we shipped installing from API. The description information is available in the API so we can just fetch it from there for core taps.

**Comment by SMillerDev** [comment]
While we could, I think algolia would be quicker and generally provide better results

**Comment by MikeMcQuaid** [comment]
> I think algolia would be quicker and generally provide better results

If it's quicker/better: let's do it.

File: Library/Homebrew/search.rb [hunk 1]
```diff
@@ -20,13 +24,32 @@ def self.query_regexp(query)
 
     def self.search_descriptions(string_or_regex, args, search_type: :desc)
       both = !args.formula? && !args.cask?
-      eval_all = args.eval_all? || Homebrew::EnvConfig.eval_all?
+      output, _, status = curl_output("--silent",
+                                      "--header", "Content-Type: application/json; charset=UTF-8",
+                                      "--header", "X-Algolia-API-Key:#{ALGOLIA_API_KEY}",
+                                      "--header", "X-Algolia-Application-Id:#{ALGOLIA_APPLICATION_ID}",
+                                      "--location", "https://#{ALGOLIA_APPLICATION_ID}-dsn.algolia.net/1/indexes/#{ALGOLIA_INDEX}?query=#{string_or_regex}")
```
Comment (carlocab) [inline Library/Homebrew/search.rb:31]: Not sure feeding a regex here works?

**Comment by Bo98** [comment]
> I think algolia would be quicker and generally provide better results

Quicker: unlikely since it's requires a network request. Better results: perhaps.

**Comment by carlocab** [comment]
> Quicker: unlikely since it's requires a network request. Better results: perhaps.

Seems better to me to avoid a network request if we can help it, even if the price we pay is worse results. For the record: I suspect the results will not be better if we handle searching locally correctly.

**Pull request closed without merging**